### PR TITLE
Issue #12: implement DataSourceWithLogsContextSupport interface

### PR DESCRIPTION
**Pull request by idrissneumann**

### CODE REVIEW

**Comment by idrissneumann** [comment]
For the GA compatibility check, it'll be handled with this issue in another PR: https://github.com/quickwit-oss/quickwit-datasource/issues/24

File: package.json [hunk 1]
```diff
@@ -61,10 +61,10 @@
   },
   "dependencies": {
     "@emotion/css": "^11.1.3",
-    "@grafana/data": "9.5.1",
-    "@grafana/experimental": "^1.1.0",
-    "@grafana/runtime": "9.5.1",
-    "@grafana/ui": "9.5.1",
+    "@grafana/data": "10.2.2",
```
Comment (fmassot) [inline package.json:64]: Can we do the package ugprade in a separate PR?

File: src/datasource.ts [hunk 2]
```diff
@@ -740,3 +819,36 @@ function luceneEscape(value: string) {
 
   return value.replace(/([\!\*\+\-\=<>\s\&\|\(\)\[\]\{\}\^\~\?\:\\/"])/g, '\\$1');
 }
+
+function createContextTimeRange(rowTimeEpochMs: number, direction: string, intervalPattern: Interval | undefined) {
+  const offset = 7;
+  // For log context, we want to request data from 7 subsequent/previous indices
+  if (intervalPattern) {
```
Comment (fmassot) [inline src/datasource.ts:826]: we don't need the `intervalPattern`, this is related to es index pattern and we don't support that.

File: package.json (hunk 1 above)
Comment (idrissneumann) [reply package.json]: Yes, I reverted this :)

File: src/datasource.ts (hunk 2 above)
Comment (idrissneumann) [reply src/datasource.ts]: Done

File: pkg/quickwit/client/client.go [hunk 3]
```diff
@@ -166,6 +166,13 @@ func (c *baseClientImpl) ExecuteMultisearch(r *MultiSearchRequest) (*MultiSearch
 
 	c.logger.Debug("Received multisearch response", "code", res.StatusCode, "status", res.Status, "content-length", res.ContentLength)
 
+	if res.StatusCode >= 400 {
```
Comment (fmassot) [inline pkg/quickwit/client/client.go:169]: We should return an error immediately.
We can improve that in a dedicated PR.

File: src/IntervalMap.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { DurationUnit } from '@grafana/data';
```
Comment (fmassot) [inline src/IntervalMap.ts:1]: do we need this interval map?

**Review by fmassot — APPROVED** [review]
Can you check if there are some tests that we can retrieve from the elasticsearch plugin? 
it seems we're a bit light on this.

I put a few minor comments, can you fix that before merging?

File: src/IntervalMap.ts (hunk 4 above)
Comment (idrissneumann) [reply src/IntervalMap.ts]: Done, don't need it anymore indeed :)

File: pkg/quickwit/client/client.go (hunk 3 above)
Comment (idrissneumann) [reply pkg/quickwit/client/client.go:169]: Issue created: https://github.com/quickwit-oss/quickwit-datasource/issues/29

**Comment by idrissneumann** [comment]
> Can you check if there are some tests that we can retrieve from the elasticsearch plugin?
it seems we're a bit light on this.

Yeah there's more asserts on the elastic plugin but the code is diverging, there's some typing/object structure divergences:

Screenshot 2023-11-24 at 10 08 57
